logx/options.go

```diff
@@ -0,0 +1,41 @@
+package logx
+
+import "io"
+
+// Option is the common type of functions that set options
+type Option func(*options)
```

Comment:
if you make it unexported, then golint will warn that the `With*` constructors return an unexported type and will fail.